ui/src/app/cluster-workflow-templates/components/cluster-workflow-template-details/cluster-workflow-template-details.tsx

```diff
@@ -30,6 +30,14 @@ export const ClusterWorkflowTemplateDetails = ({history, location, match}: Route
     const [template, setTemplate] = useState<ClusterWorkflowTemplate>();
     const [edited, setEdited] = useState(false);
 
+    useEffect(() => {
+        return history.listen(newLocation => {
```

Comment:
`listen` returns an unlistener that you need to invoke in the cleanup function for `useEffect` or this will leak memory (true in all of the other spots as well).